### PR TITLE
[prometheus-node-exporter] Improve and correct podLabels, add commonLabels

### PR DESCRIPTION
**Pull request by sarnowski-defense-unicorns**
#### What this PR does / why we need it

Update `.Values.podLabels` templating in `prometheus-node-exporter.labels` helper function to use `tpl`. This allows for more flexibility with extra labels, e.g.

```
podLabels:
  version: "{{ .Chart.AppVersion }}"
```

#### Which issue this PR fixes

N/A

#### Special notes for your reviewer

N/A

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by sarnowski-defense-unicorns** [comment]
@gianrubio @zanhsieh @zeritti please approve pending workflow actions.

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @sarnowski-defense-unicorns, for your PR. Please, add a test case for the field in the `ci`  directory.

File: charts/prometheus-node-exporter/templates/_helpers.tpl [hunk 1]
```diff
@@ -44,7 +44,7 @@ app.kubernetes.io/part-of: {{ include "prometheus-node-exporter.name" . }}
 app.kubernetes.io/version: {{ . | quote }}
 {{- end }}
 {{- with .Values.podLabels }}
```
Comment (zeritti) [inline charts/prometheus-node-exporter/templates/_helpers.tpl:46]: I have just noticed that having podLabels in here is wrong - `podLabels` are to be attached to pods only but through this helper get attached to all resources. Would you be willing to correct it? We could rename the field to `commonLabels`, keep it here and make the field `commonLabels` in values. `podLabels` should then be inserted in the daemonset pod template.

Comment (sarnowski-defense-unicorns) [reply charts/prometheus-node-exporter/templates/_helpers.tpl:46]: Was out last week, will take a look today or tomorrow.

**Review by zanhsieh — APPROVED** [review]

**Review by zeritti — APPROVED** [review]

**Pull request merged**